src/service/extensions-impl.js

```diff
@@ -572,6 +572,12 @@ export class Extensions {
     scriptElement.setAttribute('data-script', extensionId);
     scriptElement.setAttribute('i-amphtml-inserted', '');
 
+    // Propagate nonce to all generated script tags.
+    const currentScript = this.win.document.querySelector('script[nonce]');
```

Comment:
This should be restricted to `document.head` since nonce is also allowed elsewhere e.g. JSON scripts in the body.

Also, do we use this codepath in PWA (shadow AMP) mode? If so, we should minimally be more specific in the query selector to avoid picking up the shell app's scripts.

Reply:
> This should be restricted to document.head since nonce is also allowed elsewhere e.g. JSON scripts in the body.

Is that true? Doesn't the nonce need to be identical everywhere on the page?

> Also, do we use this codepath in PWA (shadow AMP) mode? If so, we should minimally be more specific in the query selector to avoid picking up the shell app's scripts.

I'll investigate. Worse case scenario sounds like it'll be an extra unnecessary property as opposed to breaking functionality.

Reply:
> Is that true? Doesn't the nonce need to be identical everywhere on the page?

Here's a contrived example:

```html
<amp-list>
  <template type="amp-mustache">
    <script type="application/json" nonce="{{foo}}">[]</script>
  </template>
</amp-list>
```

But as far as CSP is concerned, I think there's no functional difference. 

IMO we should still try to "propagate nonce from parent script to child script" rather than "copy the first nonce we find". It's more precise and would prevent confusing scenarios like the one above.